taipy/gui/_renderers/builder.py

```diff
@@ -483,6 +483,23 @@ def _get_dataframe_attributes(self) -> "_Builder":
         date_format = _add_to_dict_and_get(self.__attributes, "date_format", "MM/dd/yyyy")
         data = self.__attributes.get("data")
         data_hash = self.__hashes.get("data", "")
```

Comment:
This could be extracted into a function since the logic itself is self-contained.  we can also do early return if it is extracted into a separate function

Reply:
can't do as cmp_hash is used in the code